ibis/backends/dask/client.py

```diff
@@ -0,0 +1,468 @@
+"""The dask client implementation."""
+
+from __future__ import absolute_import
+
+import re
+from functools import partial
+
+import dask.dataframe as dd
+from dask.dataframe.core import DataFrame
+import dateutil.parser
+import numpy as np
+import pandas as pd
+import pytz
+import toolz
+from multipledispatch import Dispatcher
+from pandas.api.types import CategoricalDtype, DatetimeTZDtype
+from pkg_resources import parse_version
+
+import ibis.client as client
+import ibis.common.exceptions as com
+import ibis.expr.datatypes as dt
+import ibis.expr.operations as ops
+import ibis.expr.schema as sch
+import ibis.expr.types as ir
+
+from .core import execute_and_reset
+
+try:
+    infer_dask_dtype = pd.api.types.infer_dtype
+except AttributeError:
+    infer_dask_dtype = pd.lib.infer_dtype
+
+
+_ibis_dtypes = toolz.valmap(
+    np.dtype,
+    {
+        dt.Boolean: np.bool_,
+        dt.Null: np.object_,
+        dt.Array: np.object_,
+        dt.String: np.object_,
+        dt.Binary: np.object_,
+        dt.Date: 'datetime64[ns]',
+        dt.Time: 'timedelta64[ns]',
+        dt.Timestamp: 'datetime64[ns]',
+        dt.Int8: np.int8,
+        dt.Int16: np.int16,
+        dt.Int32: np.int32,
+        dt.Int64: np.int64,
+        dt.UInt8: np.uint8,
+        dt.UInt16: np.uint16,
+        dt.UInt32: np.uint32,
+        dt.UInt64: np.uint64,
+        dt.Float32: np.float32,
+        dt.Float64: np.float64,
+        dt.Decimal: np.object_,
+        dt.Struct: np.object_,
+    },
+)
+
+
+_numpy_dtypes = toolz.keymap(
```

Comment:
hmm i think we have this elsewhere no?

Reply:
Another thing that is directly cloned from the pandas backend. I can change to import from there?

Reply:
yes, go ahead an import things from pandas directly (if its more of 'shared' code then we can likley find a better place), but easy enough to just import now.